python/pyarrow/tests/test_udf.py

```diff
@@ -0,0 +1,501 @@
+                          [
+                              pa.array([10, 20], pa.int64()),
+                              pa.array([2, 4], pa.int64()),
+                              pa.array([5, 10], pa.int64())
+                          ],
+                          mock_udf_context()
+                          )
+
+
+def test_scalar_udf_array_varargs(varargs_func_fixture):
+    check_scalar_function(varargs_func_fixture,
+                          [
+                              pa.array([2, 3], pa.int64()),
+                              pa.array([10, 20], pa.int64()),
+                              pa.array([3, 7], pa.int64()),
+                              pa.array([20, 30], pa.int64()),
+                              pa.array([5, 10], pa.int64())
+                          ],
+                          mock_udf_context
+                          )
+
+
+def test_registration_errors():
+    # validate function name
+    doc = {
+        "summary": "test udf input",
+        "description": "parameters are validated"
+    }
+    in_types = {"scalar": pa.int64()}
+    out_type = pa.int64()
+
+    def test_reg_function(context):
+        return pa.array([10])
+
+    with pytest.raises(TypeError):
+        pc.register_scalar_function(test_reg_function,
+                                    None, doc, in_types,
+                                    out_type)
+
+    # validate function
+    with pytest.raises(TypeError, match="func must be a callable"):
+        pc.register_scalar_function(None, "test_none_function", doc, in_types,
+                                    out_type)
+
+    # validate output type
+    expected_expr = "DataType expected, got <class 'NoneType'>"
+    with pytest.raises(TypeError, match=expected_expr):
+        pc.register_scalar_function(test_reg_function,
+                                    "test_output_function", doc, in_types,
+                                    None)
+
+    # validate input type
+    expected_expr = r'in_types must be a dictionary of DataType'
+    with pytest.raises(TypeError, match=expected_expr):
+        pc.register_scalar_function(test_reg_function,
+                                    "test_input_function", doc, None,
+                                    out_type)
+
+    # register an already registered function
+    # first registration
+    pc.register_scalar_function(test_reg_function,
+                                "test_reg_function", doc, {},
+                                out_type)
+    # second registration
+    expected_expr = "Already have a function registered with name:" \
+        + " test_reg_function"
+    with pytest.raises(pa.lib.ArrowKeyError, match=expected_expr):
+        pc.register_scalar_function(test_reg_function,
+                                    "test_reg_function", doc, {},
+                                    out_type)
+
+
+def test_varargs_function_validation(varargs_check_func_fixture):
+    _function, func_name = varargs_check_func_fixture
```

Comment:
```suggestion
    _, func_name = varargs_check_func_fixture
```
is what's usually done to ignore things in unpacking